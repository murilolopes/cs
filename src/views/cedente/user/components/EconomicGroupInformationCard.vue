<template>
  <b-card :border-variant="borderVariant" :class="{ 'pending-documentation-bg-color': documentationInProgress }">
    <div class="d-flex flex-row justify-content-between align-items-start w-100">
      <div class="d-flex flex-column align-items-start justify-content-start w-50">
        <h3 class="mb-0" v-if="cedente.nome_fantasia.length > 20" tabindex="0" v-b-tooltip.hover :title="cedente.nome_fantasia">{{ truncateCedenteName }}</h3>
        <h3 class="mb-0" v-else>{{ cedente.nome_fantasia }}</h3>
        <p class="lead text-dark">{{ cedente.cnpj.replace(/^(\d{2})(\d{3})(\d{3})(\d{4})(\d{2})/, '$1.$2.$3/$4-$5') }}</p>
        <p class="lead text-muted">{{ subsidiarySizeLabel }}</p>
      </div>

      <div class="d-flex flex-row align-items-start justify-content-end w-50">
        <div class="d-flex flex-column justify-content-between align-items-end gap-1">
          <div class="d-flex flex-row align-items-center">
            <feather-icon :icon="badgeStatus.icon" size="26" :class="`text-${badgeStatus.variant}`" />
            <b-badge :variant="`light-${badgeStatus.variant}`" :class="`border-${badgeStatus.variant} ml-1`">{{
              badgeStatus.text }}</b-badge>
          </div>
          <div class="d-flex flex-row align-items-center">
            <b-button variant="primary" v-if="showAnalisyButton" @click="goToCedentePage">{{ buttonLabel }}</b-button>
          </div>
        </div>
      </div>
    </div>
  </b-card>
</template>

<script>
import {
  BCard,
  BBadge,
  BButton,
  VBTooltip
} from "bootstrap-vue";

export default {
  name: "EconomicGroupInformationCard",
  components: {
    BCard,
    BBadge,
    BButton,
  },
  props: {
    cedente: {
      type: Object,
      required: true,
      default: () => ({
        status: "nao_enviado",
        filiais: [],
      }),
    },
  },
  directives: {
    'b-tooltip': VBTooltip,
  },
  methods: {
    goToCedentePage() {
      const routeName = this.userType === 'cedente' ? 'complete-economic-group-registration' : 'admin-economicGroup.general-data';
      const params = this.userType === 'cedente' ? { id: this.cedente.id } : { id: this.cedente.grupo_economico_id, idMatriz: this.cedente.id };

      console.log(params)

      this.$router.push({ name: routeName, params });
    },
  },
  computed: {
    documentationInProgress() {
      return !['aprovado', 'reprovado'].includes(this.cedente.status_documentacao);
    },
    userType() {
      return this.$store.state.auth.userData.user_type
    },
    buttonLabel() {
      const label = {
        cedente: 'Preencher informações agora',
        admin: 'Analisar informações',
      }
      return label[this.userType]
    },
    showAnalisyButton() {
      const status = {
        cedente: {
          nao_enviado: true,
          em_validacao: false,
          aprovado: false,
          reprovado: true,
          blocked: false,
        },
        admin: {
          nao_enviado: true,
          em_validacao: true,
          aprovado: false,
          reprovado: true,
          blocked: true,
        },
      }

      return status[this.userType][this.cedente.status_documentacao]
    },
    subsidiarySizeLabel() {
      if (this.cedente.filiais?.length === 0) return "Sem filiais";

      return `+${this.cedente.filiais?.length} ${this.cedente.filiais?.length > 1 ? "Filiais incluídas" : "Filial incluída"}`;
    },
    borderVariant() {
      const status = {
        em_validacao: "light",
        nao_enviado: "light",
        reprovado: "danger",
        aprovado: "success",
        blocked: "light",
      }

      return status[this.cedente.status_documentacao]
    },
    badgeStatus() {
      const status = {
        cedente: {
          nao_enviado: {
            icon: 'AlertCircleIcon',
            variant: 'warning',
            text: 'Aguardando seu preenchimento',
          },
          em_validacao: {
            icon: 'ClockIcon',
            variant: 'warning',
            text: 'Informações em análise',
          },
          aprovado: {
            icon: 'CheckCircleIcon',
            variant: 'success',
            text: 'Aprovado',
          },
          reprovado: {
            icon: 'AlertOctagonIcon',
            variant: 'danger',
            text: 'Pendência no cadastro',
          },
          blocked: {
            icon: 'XCircleIcon',
            variant: 'dark',
            text: 'Conta temporariamente bloqueada',
          },
        },
        admin: {
          nao_enviado: {
            icon: 'AlertCircleIcon',
            variant: 'warning',
            text: 'Informações incompletas',
          },
          em_validacao: {
            icon: 'ClockIcon',
            variant: 'warning',
            text: 'Aguardando análise',
          },
          aprovado: {
            icon: 'CheckCircleIcon',
            variant: 'success',
            text: 'Aprovado',
          },
          reprovado: {
            icon: 'AlertOctagonIcon',
            variant: 'danger',
            text: 'Reprovado',
          },
          blocked: {
            icon: 'XCircleIcon',
            variant: 'dark',
            text: 'Oferta desativada',
          },
        }
      }

      return status[this.userType][this.cedente.status_documentacao]
    },
    truncateCedenteName() {
      return this.cedente.nome_fantasia.slice(0, this.cedente.nome_fantasia.length - 10).trim() + '...'
    },
  },
};
</script>

<style lang="scss">
.pending-documentation-bg-color {
  background-color: #F8FAFC !important;
}

.gap-1 {
  gap: 1.5rem;
}
</style>
