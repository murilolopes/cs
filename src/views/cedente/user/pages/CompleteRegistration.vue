<template>
  <div>
    <b-card>
      <div class="d-flex flex-row justify-content-between align-items-start">
        <div class="d-flex flex-column align-items-start mb-1 p-0">
          <p class="h3 mb-0">Grupo econômico</p>
          <h3 class="text-dark mb-2">{{ economicGroupName }}</h3>
          <span class="bs-height-lg">Preencha e envie as informações referentes à cada Matriz registrada</span>
        </div>
        <div class="d-flex flex-row align-items-center justify-content-end">
          <b-button variant="outline-primary" @click="$router.push({
            name: 'economic-group.cedentes.new'
          })">Incluir CNPJ +</b-button>
        </div>
      </div>

      <b-row align-v="start">
        <b-col sm="12" md="6" v-for="cedente in cedentes" :key="cedente.id">
          <economic-group-information-card :cedente="cedente" />
        </b-col>
      </b-row>
    </b-card>
  </div>
</template>

<script>
import {
  BCard,
  BRow,
  BCol,
  BButton,
} from "bootstrap-vue";
import EconomicGroupInformationCard from "@/views/cedente/user/components/EconomicGroupInformationCard";

export default {
  name: "CompleteRegistration",
  components: {
    BCard,
    BRow,
    BCol,
    BButton,
    EconomicGroupInformationCard,
  },
  data() {
    return {
      economicGroupName: '',
      cedentes: []
    };
  },
  async mounted() {
    try {
      const { data } = await this.$store.dispatch("auth/getCurrentEconomicGroupData");
      this.economicGroupName = data.nome
      this.cedentes = data.cedentes
    } catch (error) {
      console.log(error);
    }
  },
};
</script>
