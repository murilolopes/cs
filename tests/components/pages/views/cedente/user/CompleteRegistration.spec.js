import Vuex from "vuex";
import { mount, shallowMount, createLocalVue } from "@vue/test-utils";
import CompleteRegistration from '@/views/cedente/user/pages/CompleteRegistration.vue'
import EconomicGroupInformationCard from '@/views/cedente/user/components/EconomicGroupInformationCard.vue'

import flushPromises from 'flush-promises'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('CompleteRegistration.vue', () => {
  it("Should render correctly", () => {
    let store = new Vuex.Store({
      modules: {
        cedente: {
          namespaced: true,
          actions: {
            getEconomicGroups: jest.fn(() => Promise.resolve([]))
          }
        }
      }
    });

    const wrapper = shallowMount(CompleteRegistration, { localVue, store })
    expect(wrapper.element).toMatchSnapshot();
  });

  it("Should render EconomicGroupInformationCards correctly", async () => {
    const response = {
      data: [{
        name: 'Teste 1',
        cnpj: '12345678901234',
        status: 'active',
        filiais: [{id:1}, {id:2}, {id:3}]
      }, {
        name: 'Teste 2',
        cnpj: '12345678901234',
        status: 'active',
        filiais: [{id:1}, {id:2}]
      }, {
        name: 'Teste 3',
        cnpj: '12345678901234',
        status: 'active',
        filiais: [{id:1}]
      }]
    }

    let store = new Vuex.Store({
      modules: {
        cedente: {
          namespaced: true,
          actions: {
            getEconomicGroups: jest.fn()
          }
        }
      }
    });

    store.dispatch = jest.fn(() => Promise.resolve(response))

    const wrapper = await mount(CompleteRegistration, { localVue, store })
    expect(wrapper.vm.$store.dispatch).toHaveBeenCalledWith('cedente/getEconomicGroups')
    expect(wrapper.vm.economicGroups).toEqual(response.data)

    await flushPromises()

    const economicGroupInformationCards = wrapper.findAllComponents(EconomicGroupInformationCard)
    expect(economicGroupInformationCards).toHaveLength(wrapper.vm.economicGroups.length)
  })
})
