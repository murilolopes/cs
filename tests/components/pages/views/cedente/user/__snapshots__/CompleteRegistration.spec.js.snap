// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompleteRegistration.vue Should render correctly 1`] = `
<div>
  <b-card-stub
    bodytag="div"
    footertag="div"
    headertag="div"
    subtitletag="h6"
    subtitletextvariant="muted"
    tag="div"
    titletag="h4"
  >
    <div
      class="d-flex flex-row justify-content-between align-items-start"
    >
      <div
        class="d-flex flex-column align-items-start mb-1 p-0"
      >
        <p
          class="h3 mb-0"
        >
          Grupo econômico
        </p>
         
        <h3
          class="text-dark mb-2"
        >
          
        </h3>
         
        <span
          class="bs-height-lg"
        >
          Preencha e envie as informações referentes à cada Matriz registrada
        </span>
      </div>
       
      <div
        class="d-flex flex-row align-items-center justify-content-end"
      >
        <b-button-stub
          tag="button"
          target="_self"
          type="button"
          variant="outline-primary"
        >
          Incluir CNPJ +
        </b-button-stub>
      </div>
    </div>
     
    <b-row-stub
      alignv="start"
      tag="div"
    />
  </b-card-stub>
</div>
`;
